File: Medium/0080-remove-duplicates-from-sorted-array-ii.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        const int N = nums.size();
        int i = 0;
        int cnt = 0;
        for (int j = 0; j < N; j++) {
            if (j > 0 && nums[j] == nums[j - 1]) {
                ++cnt;
                if (cnt > 2) continue;
            } else {
                cnt = 1;
            }
            nums[i++] = nums[j];
        }
        return i;
    }
};

File: Algorithms/Array/0080-remove-duplicates-from-sorted-array-ii.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i = cnt = 0
        for j in range(len(nums)):
            if j > 0 and nums[j] == nums[j - 1]:
                cnt += 1
            else:
                cnt = 1
            if cnt <= 2:
                nums[i] = nums[j]
                i += 1
        return i
